Amazon.IonDotnet.Tests/Internals/ReaderLocalTableTest.cs

```diff
@@ -48,5 +56,242 @@ void assertTable(ISymbolTable tab, params (string sym, int id)[] syms)
+            CheckSymbol("s11", systemMaxId + 1, appended);
+            CheckSymbol("o1", systemMaxId + 2, original);
+
+            CheckSymbol("s11", systemMaxId + 1, appended);
+            CheckSymbol("s21", systemMaxId + 2, appended);
+            CheckSymbol("a1", systemMaxId + 3, appended);
+
+            // new symbols in `original` are not accessible from `appended` after import
+            Assert.IsNull(original.Find("a1"));
+            Assert.IsNull(original.Find("o1"));
+        }
+
+
+        private static ISymbolTable OneValue(string text)
+        {
+            IIonReader reader = IonReaderBuilder.Build(text);
+            reader.MoveNext();
+            return reader.GetSymbolTable();
+        }
+
+        private static void CheckLocalTable(ISymbolTable symbolTable)
+        {
+            Assert.IsTrue(symbolTable.IsLocal);
+            Assert.IsFalse(symbolTable.IsShared);
+            Assert.IsFalse(symbolTable.IsSystem);
+            Assert.IsFalse(symbolTable.IsSubstitute);
+            Assert.IsNotNull(symbolTable.GetImportedTables());
+
+            CheckUnknownSymbol(" not defined ", UNKNOWN_SYMBOL_ID, symbolTable);
+
+            ISymbolTable systemTable = symbolTable.GetSystemTable();
+            CheckSystemTable(systemTable);
+            Assert.AreEqual(systemTable.IonVersionId, symbolTable.IonVersionId);
+        }
+
+        private static void CheckSystemTable(ISymbolTable symbolTable)
+        {
+            Assert.IsFalse(symbolTable.IsLocal);
+            Assert.IsTrue(symbolTable.IsShared);
+            Assert.IsTrue(symbolTable.IsSystem);
+            Assert.IsFalse(symbolTable.IsSubstitute, "table is substitute");
+            Assert.AreSame(symbolTable, symbolTable.GetSystemTable());
+            Assert.AreEqual(SystemSymbols.Ion10MaxId, symbolTable.MaxId);
+            Assert.AreEqual(ION_1_0, symbolTable.IonVersionId);
+        }
+
+        private static void CheckSymbol(string text, int sid, ISymbolTable symbolTable)
+        {
+            CheckSymbol(text, sid, false, symbolTable);
+        }
+
+        private static void CheckUnknownSymbol(string text, int sid, ISymbolTable symbolTable)
+        {
+            CheckUnknownSymbol(text, symbolTable);
+
+            if (sid != UNKNOWN_SYMBOL_ID)
+            {
+                CheckUnknownSymbol(sid, symbolTable);
+            }
+        }
+
+        private static void CheckUnknownSymbol(string text, ISymbolTable symbolTable)
+        {
+            Assert.AreEqual(default, symbolTable.Find(text));
+            Assert.AreEqual(UNKNOWN_SYMBOL_ID, symbolTable.FindSymbolId(text));
+            if (symbolTable.IsReadOnly)
+            {
+                try
+                {
+                    symbolTable.Intern(text);
+                    Assert.Fail("Expected exception");
+                }
+                catch (NotSupportedException)
+                {
```

Comment:
If the test goes into here, should it fail? If so, can we assert the fail in here?

Reply:
This is what it looks like in Java:

`    public static void checkUnknownSymbol(String text, SymbolTable symtab)
    {
        assertEquals(null, symtab.find(text));
        assertEquals(UNKNOWN_SYMBOL_ID, symtab.findSymbol(text));
        if (symtab.isReadOnly())
        {
            try {
                symtab.intern(text);
                fail("Expected exception");
            }
            catch (ReadOnlyValueException e) { }
        }
    }`